modules/react-fiber/src/reconciler.ts

```diff
@@ -0,0 +1,304 @@
+    log.info('createInstance');
+    log.info('createInstance :: type: %s', type);
+    log.info('createInstance :: props: %O', props);
+    log.info('\n');
+
+    const name = toPascal(type);
+
+    if (!catalogue[name]) {
+      throw new Error(`Unsupported element type: ${type}`);
+    }
+
+    const instance: Instance = new catalogue[name](props);
+
+    if (isView(instance)) {
+      viewCache.set(instance.id, instance);
+      filterCache.set(instance.id, []);
+
+      const state = root.getState();
+
+      // Set view immediately so we can use during layer appends
+      state.deckgl.setProps({
+        views: Array.from(viewCache.values())
+      });
+    } else {
+      // NOTE: View classes cannot be extended and or have properties added to it
+      instance.__deckglfiber = {
+        root,
+        view: undefined
+      };
+    }
+
+    return instance;
+  },
+
+  appendChild,
+
+  appendInitialChild: appendChild,
+
+  // @ts-expect-error "parent" type doesn't matter
+  appendChildToContainer: appendChild,
+
+  removeChild,
+
+  // @ts-expect-error "parent" type doesn't matter
+  removeChildFromContainer: removeChild,
+
+  insertBefore,
+
+  // @ts-expect-error "parent" type doesn't matter
+  insertInContainerBefore: insertBefore,
+
+  finalizeInitialChildren: falseReturn,
+
+  // TODO: we could look at cleaning up deckgl here with the caveat that
+  // deck.finalize() is already called in the React component cleanup.
+  clearContainer: falseReturn,
+
+  // NOTE: may need to attach handlers here
+  commitMount: noop,
+
+  // NOTE: this fires per instance, since deckgl cannot update individual layers
+  // atomically we skip doing any work in this function
+  commitUpdate: noop,
+
+  // NOTE: since deck.gl already does extensive diffing, we just need to construct
+  // a new object to replace the prior. "prepareUpdate" appears to fire twice in strict mode.
+  prepareUpdate(oldInstance, type, oldProps, newProps) {
+    log.info('prepareUpdate');
+    log.info('createInstance :: oldInstance: %O', oldInstance);
+    log.info('createInstance :: type: %s', type);
+    log.info('createInstance :: oldProps: %O', oldProps);
+    log.info('createInstance :: newProps: %O', newProps);
+    log.info('\n');
+
+    const name = toPascal(type);
+    const instance: Instance = new catalogue[name](newProps);
+    const currentIndex = layerCache.findIndex(v => v.id === instance.id);
+
+    layerCache[currentIndex] = instance;
+
+    return true;
+  },
+
+  prepareForCommit: nullReturn,
+
+  resetAfterCommit(container: Store) {
+    log.info('resetAfterCommit');
+
+    const state = container.getState();
+
+    /**
+     * TODO: implicit view/layer filter logic
+     * const existingLayerFilter = state.deckgl.props.layerFilter;
+     * const hasDeclarativeViews = viewCache.size > 0;
+     */
+
+    state.deckgl.setProps({
+      // TODO: investigate why we need to spread the array here for deckgl to pick up layer changes
```

Comment:
My hunch here is that Deck does some internal shallow compare against the layer list array? e.g. doing:

```
state.deckgl.setProps({
  layers: layerCache,
});
```

Does not work since we are mutating `layerCache` directly in some of the other functions